cmd/kubeapps-apis/plugins/helm/packages/v1alpha1/server.go

```diff
@@ -1024,3 +996,75 @@ func (s *Server) GetInstalledPackageResourceRefs(ctx context.Context, request *c
 		}, nil
 	}
 }
+
+func (s *Server) AddPackageRepository(ctx context.Context, request *corev1.AddPackageRepositoryRequest) (*corev1.AddPackageRepositoryResponse, error) {
+	log.Infof("+helm AddPackageRepository [%v]", request)
+	if request == nil {
+		return nil, status.Errorf(codes.InvalidArgument, "no request provided")
+	}
+	if request.Context == nil || request.Context.Namespace == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "no request Context namespace provided")
+	}
+	cluster := request.GetContext().GetCluster()
+	if cluster != "" && cluster != s.kubeappsCluster {
+		return nil, status.Errorf(
+			codes.Unimplemented,
+			"not supported yet: request.Context.Cluster: [%v]",
+			request.Context.Cluster)
+	}
+
+	if request.Name == "" {
+		return nil, status.Errorf(codes.InvalidArgument, "no package repository Name provided")
+	}
+	name := types.NamespacedName{Name: request.Name}
+	if request.GetNamespaceScoped() {
```

Comment:
i think we have agreed to validate that the namespace and scope were consistent (e.g. if scope is global but namespace is not the global namespace, then it should be a validation error)